### PR TITLE
Add WeightedHuberLoss and per-loss delta for Huber losses

### PR DESCRIPTION
**Pull request by ARPeeketi**
Adds WeightedHuberLoss, following the pattern of WeightedMSELoss and WeightedMAELoss. The weighted module holds torch.nn.HuberLoss(reduction="none") for the huber part and only adds the same weight normalization the other weighted losses use. Both HuberLoss and WeightedHuberLoss take delta (default 1.0, matching torch) — delta is a per-term setting, I train energies and forces with different deltas — and the plain HuberLoss node is built from torch.nn.HuberLoss with the AutoKw mixin. With the default arguments nothing changes for existing users.

Tests are in tests/test_losses.py.

### CODE REVIEW

**Comment by lubbersnick** [comment]
It's functional code, but it will be cleaned up a bit if you use the [HuberLoss module](https://docs.pytorch.org/docs/2.13/generated/torch.nn.HuberLoss.html) along with the AutoKw mixin for instantiation. See for example [LPReg](https://github.com/lanl/hippynn/blob/development/hippynn/graphs/nodes/loss.py#L163) (or many other AutoKw instances)

**Comment by ARPeeketi** [comment]
Good point — switched both nodes to AutoKw with torch.nn.HuberLoss and the weighted module, following the LPReg pattern. Cleaner, thanks.

File: hippynn/layers/algebra.py [hunk 1]
```diff
@@ -47,6 +47,20 @@ class WeightedMAELoss(_WeightedLoss):
     loss_func = staticmethod(torch.nn.functional.l1_loss)
 
 
+class WeightedHuberLoss(_WeightedLoss):
```
Comment (lubbersnick) [inline hippynn/layers/algebra.py:50]: I don't think you need to define this, I think you just need to import this at the node definition:

https://docs.pytorch.org/docs/2.13/generated/torch.nn.HuberLoss.html

**Comment by ARPeeketi** [comment]
Reworked it — the plain node now builds torch.nn.HuberLoss directly through AutoKw with delta. For the weighted one: delta is a per-term setting (my force term uses 5, energy 1) and the weights are per-sample, so the module takes delta, holds torch.nn.HuberLoss(reduction="none") for the huber part, and only defines the same weight normalization WeightedMSELoss and WeightedMAELoss use.
